app/src/main/java/edu/usf/sas/pal/muser/firebase/models/SongAction.kt

```diff
@@ -0,0 +1,22 @@
+package edu.usf.sas.pal.muser.firebase.models
+
+/**
+ * A model class that stores the actions performed on the song track. The actions like play, pause,
+ * skip, repeat and seek are stored along with the nested [SongData] class.
+ */
+
+data class SongAction
```

Comment:
Let's rename this to `Event` - thinking more, some events we want to capture might not have an explicit action associated with them.

Reply:
Understood. I'm guessing these events can be switching to the next song automatically or the song repeating automatically if there isn't any song to play from the queue?

Reply:
I think in the event enum we'll actually want two events to distinguish between manual and automated events - maybe something like `PLAY_MANUAL` and `PLAY`, or something similar. This may be a little tricky to capture if Android itself doesn't differentiate between the two, as the manual would need to originate from a UI action and the automated would need to come from a callback. We can talk more about this when we start implementing the tracking.

Reply:
An example - you can put a song on "repeat one", which can loop the same song over and over again. We'd want to know when the user themselves initiated playing it (which would be `PLAY_MANUAL`), but we'd also want to know each time the song ended and restarted (which would be `PLAY`, and maybe an indicator that the song ended too like `END`). I'm open to thoughts exactly how this would work based on your assessment of the Shuttle code.

Reply:
You're right. What I'm thinking is, we should monitor the listeners in the `PlayerFragment.java` for capturing any `PLAY_MANUAL` events. Also, we should monitor any clicking on the list items that can play a music track in the `ViewPager` (MainActivity)  and other fragments to check for any possible selection out of the playlist. Let me know your thoughts on doing something like this.